### DISCUSSION TITLE
access input metadata added by add_input_metadata

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#17908] [Q&A]
I can add some metadata to the asset UI using the `OutputContext.add_output_metadata` method in the `handle_output` method of a custom IO manager.

Is it possible to do the same in the `load_input`  method (I would guess to the downstream tasks)?  I have tried adding that metadata using the `InputContext.add_input_metadata` method thinking that would expose the metadata to the UI but no luck.  Where does the input metadata go when you use `add_input_metadata` and where can you access this from?

While i’ve been digging I saw some discussion that said there were issues or problems with this in partitioned assets, though I am not using partitioned assets for this particular case.  There seemed to be a handful workarounds, but I’d like to understand what the `add_input_metadata` is doing and how that “should” be used to see if that would work for me before opting to create a workaround.

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1699642610081539?thread_ts=1699642610.081539&cid=C01U954MEER).*

**Comment by jamiedemaria** [comment]
I think this is a feature we will need to add. I made an issue for it here https://github.com/dagster-io/dagster/issues/17923

**Reply by Isaac-Flath** [reply]
Great thank you!  I added a correction to the issue :)

Also feature request aside, what does the current `InputContext. add_input_metadata` do? Where does the input metadata go when you use add_input_metadata and where can you access this from?  Is the fact it doesn't show up in the UI a bug? Or is it's purpose something completely different?

**Reply by sean-stach-ff** [reply]
@Isaac-Flath I can find the metadata under the input assets event list
